scenarios/command/c/command_client/main.c

```diff
@@ -173,6 +202,25 @@ int main(int argc, char* argv[])
       {
         last_command_sent_time = current_time;
 
+        proto_timestamp.seconds = current_time;
```

Comment:
Unlocking something valuable using the local time() of the requestor is generally a bad idea when the message can get blocked on the broker, etc. for long periods of time. Local time can go forward or backward in time (as NTP applies adjustments).

1. We should at least add the warning in a comment: Without additional guarantees, there are no simple ways to solve this problem. Both the requestor and the door lock must have their clocks synchronized within a specified time delta for the operation to make sense.
2. The request message should have MQTT expiryTime attached to, at least, avoid sending the request if the message expires when owned by the MQTT broker.

Reply:
Resolving with work item opened from my comment below